### PR TITLE
docs: extend: plugins: mention the sdk + systemd socket activation

### PR DESCRIPTION
**Pull request by runcom**
Mention the go-plugins-helpers project + systemd socket activation

ping @thaJeztah @calavera 

Signed-off-by: Antonio Murdaca runcom@redhat.com

### CODE REVIEW

**Comment by calavera** [comment]
LGTM

File: docs/extend/plugin_api.md [hunk 1]
```diff
@@ -96,6 +96,43 @@ directory and activates it with a handshake. See Handshake API below.
...
 they are activated only lazily, or on-demand, when they are needed.
 
+## Systemd socket activation
+
+Plugins may also be socket activated by `systemd`. The official [Plugins helpers](https://github.com/docker/go-plugins-helpers)
+natively supports socket activation. In order for a plugin to be socket activated it needs
+a `service` file and a `socket` file:
+
+```bash
+$ cat your-plugin.service
```
Comment (thaJeztah) [inline docs/extend/plugin_api.md]: small nit, but can we just put the contents of the unit file's in separate code blocks, and "your-plugin.service" outside of the code block?

File: docs/extend/plugin_api.md [hunk 2]
```diff
@@ -96,6 +96,43 @@ directory and activates it with a handshake. See Handshake API below.
...
+After=network.target your-plugin.socket
+Requires=your-plugin.socket docker.service
+
+[Service]
+ExecStart=/usr/lib/docker/your-plugin
+
+[Install]
+WantedBy=multi-user.target
+
+$ cat your-plugin.socket
```
Comment (thaJeztah) [inline docs/extend/plugin_api.md]: this would be the second code block then :+1:

**Comment by thaJeztah** [comment]
thanks! left two notes, but looks good otherwise

**Comment by runcom** [comment]
@thaJeztah fixed

**Comment by thaJeztah** [comment]
@runcom oh, I actually meant something like:

    Your `service` file (for example`/lib/systemd/system/your-plugin.service`):

    ```
    [Unit]
    Description=Your plugin
    Before=docker.service
    After=network.target your-plugin.socket
    Requires=your-plugin.socket docker.service

    [Service]
    ExecStart=/usr/lib/docker/your-plugin

    [Install]
    WantedBy=multi-user.target
    ```

    Your `socket` file (for example`/lib/systemd/system/your-plugin.socket`):

    ```
    [Unit]
    Description=Your plugin

    [Socket]
    ListenStream=/run/docker/plugins/your-plugin.sock

    [Install]
    WantedBy=sockets.target
    ```

(assuming those should be put in `/lib/systemd/system/`)

**Comment by runcom** [comment]
@thaJeztah PTAL :)

File: docs/extend/plugin_api.md [hunk 3]
```diff
@@ -96,6 +96,48 @@ directory and activates it with a handshake. See Handshake API below.
...
+## Systemd socket activation
+
+Plugins may also be socket activated by `systemd`. The official [Plugins helpers](https://github.com/docker/go-plugins-helpers)
+natively supports socket activation. In order for a plugin to be socket activated it needs
+a `service` file and a `socket` file.
+
+The `service` file (for example `/lib/systemd/system/your-plugin.service`):
+
+```bash
+$ cat your-plugin.service
```
Comment (thaJeztah) [inline docs/extend/plugin_api.md]: err, you forgot to remove the "cat ..." line (and the `bash` language-hint)

File: docs/extend/plugin_api.md [hunk 4]
```diff
@@ -96,6 +96,48 @@ directory and activates it with a handshake. See Handshake API below.
...
+[Service]
+ExecStart=/usr/lib/docker/your-plugin
+
+[Install]
+WantedBy=multi-user.target
+```
+The `socket` file (for example`/lib/systemd/system/your-plugin.socket`):
+
+```bash
+$ cat your-plugin.socket
```
Comment (thaJeztah) [inline docs/extend/plugin_api.md]: same here

(ducks)

File: docs/extend/plugin_api.md [hunk 5]
```diff
@@ -96,6 +96,48 @@ directory and activates it with a handshake. See Handshake API below.
...
+After=network.target your-plugin.socket
+Requires=your-plugin.socket docker.service
+
+[Service]
+ExecStart=/usr/lib/docker/your-plugin
+
+[Install]
+WantedBy=multi-user.target
+```
+The `socket` file (for example`/lib/systemd/system/your-plugin.socket`):
```
Comment (thaJeztah) [inline docs/extend/plugin_api.md]: oh, and missing a space after "example"

**Comment by runcom** [comment]
my bad (I'm travelling), should be fixed now

**Comment by runcom** [comment]
@thaJeztah please also take a loot as I've renamed the authorization.md file to plugins_authorization.md (to be consistent with network and volume)

**Comment by thaJeztah** [comment]
@runcom oh, nice!

Can you add 

    aliases = ["/engine/extend/authorization/"]

To the metadata section for that document (`plugins_authorization.md`) ? That should create a redirect page from the old to the new URL

**Comment by runcom** [comment]
@thaJeztah done!

File: docs/extend/plugins_authorization.md [hunk 6]
```diff
@@ -3,6 +3,7 @@
 title = "Access authorization plugin"
 description = "How to create authorization plugins to manage access control to your Docker daemon."
 keywords = ["security, authorization, authentication, docker, documentation, plugin, extend"]
+liases = ["/engine/extend/authorization/"]
```
Comment (thaJeztah) [inline docs/extend/plugins_authorization.md]: oh, copy/pasta; missing an "a"

**Comment by runcom** [comment]
@thaJeztah PTAL

**Comment by thaJeztah** [comment]
LGTM, thanks!

**Pull request merged**